--- apertium-sme-nob-0.5.0/apertium-sme-nob.sme-nob.rlx.orig	2012-06-13 11:22:44.000000000 +0200
+++ apertium-sme-nob-0.5.0/apertium-sme-nob.sme-nob.rlx	2013-06-29 23:04:00.803145461 +0200
@@ -3318,7 +3318,7 @@ SELECT:r713 LAGA IF (0 LAGA) ;
 SELECT:r714 LAHKA IF (0 LAHKA);
   	## Čujuhuvvo maiddái sierramotiivvaide dearvvašvuođalága 45. paragráfii čuolbmačilgehusaid birra jávohisvuođageatnegasvuođa ja geigenlobi ektui epikriissa oktavuođas (oasi dearvvašvuođabargiidlágas oktan mearkkašumiiguin).
 
-SELECT:r716 ("láhka") IF (0 ("lága"))(*1 ("fápmu") LINK 1 ("boahtin") BARRIER V-NOT-COP);
+SELECT:r716 ("láhka") IF (0 ("lága"))(*1 ("fápmu") LINK 1 ("boahtin"));
 
 #SELECT:r717 ("láhka") IF (0 ("lága") LINK *1 LAHKA-NOUN OR ("kriminaliseret") BARRIER V-NOT-COP);
 	## Norgga stáda lea iešguđet lágaid ja láhkaásahusaid bokte geatnegahtton.
@@ -3605,7 +3605,7 @@ SELECT:r809 ("virgi") IF (0 ("viergi" Il
     
 SELECT:r810 ("virgi") IF (0 ("viergi") LINK *-1 BOS LINK *1 ("virgáibidjat") OR ("váldobidjan") OR ("bálká")) ;    
 
-SELECT:r811 ("virgi") IF (0 ("viergi" Gen))(1 VIRGI-NOUN BARRIER NOT-ADJ) ;
+SELECT:r811 ("virgi") IF (0 ("viergi" Gen))(1 VIRGI-NOUN) ;
 	## Mii váldit Finnmárkku olggobeale virggiid goluid, maiguin ásahit dakkár gelbbolašvuođa...
   
 SELECT:r812 ("virgi") IF (*-1 ("luohpat") OR ("bissut") OR ("bidjat") BARRIER NOT-ADV)(0 ("viergi" Loc));
@@ -3614,7 +3614,7 @@ SELECT:r813 ("virgi") IF (0 ("viergi" Lo
    	## Doavttir lea máŋgii biddjon eret virggiin miehtá Suoma.
    
 SELECT:r814 ("virgi") IF (0 ("viergi" Loc))(1 COPULAS)(*2 HUMAN OR VIRGI-LOC-NOUN BARRIER NP-HEAD-NOM LINK 0 Nom); 
-SELECT:r815 ("virgi") IF (*-1 HUMAN OR VIRGI-LOC-NOUN LINK 0 Nom BARRIER NPNH OR VERB LINK 0 NP-HEAD-NOM)(0 ("viergi" Loc));  
+SELECT:r815 ("virgi") IF (*-1 HUMAN OR (VIRGI-LOC-NOUN LINK 0 Nom) BARRIER NPNH OR VERB LINK 0 NP-HEAD-NOM)(0 ("viergi" Loc));  
 	## Virggiin leat nu ollu sámegielat doaktárat go vejolaš. 
 	
 SELECT:r816 ("virgi") IF (*0 ("resursavátnivuohta") OR ("servodatsuorgi") BARRIER NPNH)(0 ("viergi" Gen))(1 ("ektui"));	
@@ -4545,7 +4545,7 @@ SELECT:r1012 N IF (-1 BOC)(1 (PUNCT LEFT
 	## Riehkit (riehkit ja fierváhirssat) mat bohtet beaŋkká siskkobeallai 
 	## olggobeal priváhta eatnamiid gullet eanaeaiggádii.
 
-REMOVE:r1013 N IF (0 Pr)(NEGATE 0 ("guovddáš"))(*1 (N Gen) OR (Pron Pers Gen) BARRIER S-BOUNDARY OR CC LINK NOT 0 RCmpnd BARRIER NPNH);
+REMOVE:r1013 N IF (0 Pr)(NEGATE 0 ("guovddáš"))(*1 (N Gen) OR (Pron Pers Gen) BARRIER S-BOUNDARY OR CC LINK NOT 0 RCmpnd);
 	## Jus divššohas sivas fysihkalaš vigiid,..  
 
 
@@ -5031,7 +5031,7 @@ SELECT:r1126 Pr IF (-1 V-MAIN LINK 0 VFI
 	## Gonagas addá láhkaásahusa lága geavaheami birra Svalbarddas.
 
 SELECT:r1129 Po IF (*-1 BOC OR CLB BARRIER NPNH OR Num)(-1 Gen)(NEGATE 0 NOUN-PO OR ("mielde"))(NEGATE 1 NP-HEAD-GEN);
-SELECT:r1130 Po IF (-1 Gen)(1 CC)(NEGATE -1 Acc LINK *-1 CC BARRIER NOT-NP LINK -1 Acc LINK -1 V-TRANS-ACT BARRIER NPNHA)(NEGATE 0 ("mielde") OR ("fárrui"));
+SELECT:r1130 Po IF (-1 Gen)(1 CC)(NEGATE -1 Acc LINK *-1 CC BARRIER NOT-NP LINK -1 Acc LINK -1 V-TRANS-ACT)(NEGATE 0 ("mielde") OR ("fárrui"));
 SELECT:r1131 Po IF (-1 Gen LINK NEGATE 0 Loc LINK *2 NP-HEAD-GEN BARRIER NPNH)(NEGATE -1 VERB OR Adv)(NEGATE 0 TIME-PREP)(1 TIME); 
 	## Dán vuostá sáhttá dadjat, ahte ...
 	## Jos sáddo deavdá buođu vuollái dahje jos sáddo hávddarda buođu vuolde... 
@@ -5374,7 +5374,7 @@ SELECT:r1214 Adv IF (-1 Acc LINK *-1 ("v
 SELECT:r1215 Adv IF (-1 Dem OR N OR A OR Pron LINK 0 Gen)(0 ("mađe"))(NEGATE *1 DADE BARRIER S-BOUNDARY);
 	## Mun bohten dan mađe árrat.		
 	
-SELECT:r1216 Adv IF (0 ("mielde"))(*-1 ("go" Pcle) LINK -1 REALCOPULAS BARRIER VFIN);
+SELECT:r1216 Adv IF (0 ("mielde"))(*-1 ("go" Pcle) LINK -1 REALCOPULAS);
 SELECT:r1217 Adv IF (0 ("mielde"))(*-1 ("leat" Qst) BARRIER VFIN);
   	## Leago dorski mielde?
   	## Lea go dorski mielde?
@@ -6154,7 +6154,7 @@ SELECT:r1429 ("visot" A Attr) IF (0 ("vi
 	
 	# viššal vs. viššalaš
 	# - - - - - - - - - -
-REMOVE:r1430 ("viššalaš") IF (*-1 HUMAN LINK 0 NP-HEAD-NOM BARRIER S-BOUNDARY);
+REMOVE:r1430 ("viššalaš") IF (*-1 (HUMAN LINK 0 NP-HEAD-NOM) BARRIER S-BOUNDARY);
   	## Sihke Mázes ja Guovdageainnus leat boarrásat viššalit finadan doaibmaguovddážiin.
 
 	# vuhtii
@@ -6209,21 +6209,21 @@ MAP:r425 (@COMP-CS←) TARGET Nom IF (*-
 MAP:r426 (@COMP-CS←) TARGET NP IF (*-1 ("go") BARRIER NOT-NP OR Nom LINK NOT -1 VFIN OR BOS)(1 CLB LINK NOT *1 VFIN BARRIER CC); 
 	## Muhto buođđogeažit ja giđđabuođut ráhkaduvvojit muđui seammá vugiid mielde go geassebuođut, ja danne heive buođđun birra obbalohkái juo muitaluvvot.
 	
-MAP:r427 (@COMP-CS←) TARGET $$ADVLCASE IF (*-1 ("go" CS) LINK 1 Indef BARRIER NOT-NP OR Nom LINK NOT -1 VFIN OR BOS LINK *-1 $$ADVLCASE BARRIER SV-BOUNDARY)(1 CLB LINK NOT *1 VFIN BARRIER CC); 
+MAP:r427 (@COMP-CS←) TARGET $$ADVLCASE IF (*-1 ("go" CS) LINK 1 Indef OR Nom LINK NOT -1 VFIN OR BOS LINK *-1 $$ADVLCASE BARRIER SV-BOUNDARY)(1 CLB LINK NOT *1 VFIN BARRIER CC); 
 
 MAP:r431 (@COMP-CS←) TARGET NP IF (*-1 ("go" CS) BARRIER NPNH LINK *-1 Nom BARRIER S-BOUNDARY)(*-1 ("go" CS) BARRIER NPNH LINK *-1 Comp BARRIER S-BOUNDARY)(NEGATE *-1 go BARRIER NPNH LINK -1 VFIN LINK *-1 VFIN BARRIER S-BOUNDARY)(0 Nom OR ADVLCASE)(1 EOS OR CLB) (NEGATE 1 CLB LINK *1 VFIN BARRIER CC) (NEGATE 1 CLB LINK *1 NP-HEAD-NOM BARRIER NPNH) (NEGATE 2 Interr LINK *1 VFIN LINK *1 VFIN BARRIER S-BOUNDARY); 
 	## ...muhto galjebut dat lea go guv-čalmmit. 
 
-MAP:r432 (@COMP-CS←) TARGET NP IF (-1 ("dego") OR ("dugo"))(0 Nom OR ADVLCASE BARRIER Nom)(NEGATE *1 VFIN-NOT-IMP OR AUX BARRIER S-BOUNDARY); 
+MAP:r432 (@COMP-CS←) TARGET NP IF (-1 ("dego") OR ("dugo"))(0 Nom OR ADVLCASE)(NEGATE *1 VFIN-NOT-IMP OR AUX BARRIER S-BOUNDARY); 
 	## Sii leat dego eŋgelat almmis.
 
 MAP:r433 (@COMP-CS←) TARGET NP IF (*-1 ("dego") OR ("dugo") BARRIER NOT-NP OR Nom)(0 Nom OR ADVLCASE)(1 EOS OR CLB);
 	## Son oahppaladdá min allagasas dego beaivváža badjáneapmi.
 
-MAP:r434 (@COMP-CS←) TARGET NP IF (*-1 ("go" CS) BARRIER NPNH LINK -1 A)(0 Nom OR ADVLCASE BARRIER Nom)(NEGATE 0 TITLE LINK 1 Prop LINK 0 Fem OR Mal OR Sur)(NEGATE 0 Attr LINK 1 N OR A)(NEGATE *1 VFIN BARRIER NOT-ADV-PCLE);
+MAP:r434 (@COMP-CS←) TARGET NP IF (*-1 ("go" CS) BARRIER NPNH LINK -1 A)(0 Nom OR ADVLCASE)(NEGATE 0 TITLE LINK 1 Prop LINK 0 Fem OR Mal OR Sur)(NEGATE 0 Attr LINK 1 N OR A)(NEGATE *1 VFIN BARRIER NOT-ADV-PCLE);
 	## Máhttájeaddji ii leat stuorit go oahpaheaddjis.
 
-MAP:r435 (@COMP-CS←) TARGET Pron IF (*-1 ("go" CS) BARRIER NPNH LINK -1 A)(0 Nom OR ADVLCASE BARRIER Nom)(NEGATE 0 Attr LINK 1 N OR A)(NEGATE *1 VFIN BARRIER NOT-ADV-PCLE);
+MAP:r435 (@COMP-CS←) TARGET Pron IF (*-1 ("go" CS) BARRIER NPNH LINK -1 A)(0 Nom OR ADVLCASE)(NEGATE 0 Attr LINK 1 N OR A)(NEGATE *1 VFIN BARRIER NOT-ADV-PCLE);
 	## Dalle dat viežžá čieža eará vuoiŋŋa vel, bahábuid go son ieš. 
 
 MAP:r436 (@COMP-CS←) TARGET Nom IF (*-2 ("dego") OR ("dugo") LINK *1 NP BARRIER NPNH LINK 1 Rel)(NEGATE 0 Attr LINK 1 N OR A)(*1 VFIN BARRIER S-BOUNDARY); 
@@ -6418,7 +6418,7 @@ REMOVE:r1441 (@CNP) IF (-1 ("dihte") LIN
     # Two transitive verbs each asking for its own object before and after the CC/CS
     ## Erenoamážit muitá borramušaid ja bohccobierggu hája ja dávttiid ruoškima go ađđamiid čiske.
     
-REMOVE:r1443 (@CVP) IF (-1 A)(0 ("go"))(1 Adv OR NP-MEMBER LINK *1 COMMA LINK 1 VFIN BARRIER NPNH);
+REMOVE:r1443 (@CVP) IF (-1 A)(0 ("go"))(1 Adv OR NP-MEMBER LINK *1 (COMMA LINK 1 VFIN) BARRIER NPNH);
     ## Dán njuolggadusa deaŧalaš eaktu lea ahte ođđa siidaosiid ásaheapmi ii galgga dagahit ahte boazolohku šaddá alit go siidii mearriduvvon lohku, geahča dás maŋŋelis.
     # specifically for comparative go-constructions
 
@@ -7396,7 +7396,7 @@ SELECT:r1697 Attr IF (0 ("vuolit"))(1 ("
 	# Comparative
 	# ===========
 
-SELECT:r1698 Comp IF (NEGATE -1 ("nu") OR ("seamma") OR CC)(NEGATE 0 (Pl Nom) LINK -1 Comp OR COMPAR LINK *-1 COPULAS BARRIER NOT-ADV-PCLE)(NEGATE 0 VFIN)(1 N LINK 1 go BARRIER NOT-N);
+SELECT:r1698 Comp IF (NEGATE -1 ("nu") OR ("seamma") OR CC)(NEGATE 0 (Pl Nom) LINK -1 Comp OR COMPAR LINK *-1 COPULAS BARRIER NOT-ADV-PCLE)(NEGATE 0 VFIN)(1 N LINK 1 go);
     	## Dážat navde Finnmárkku buoret guovlun go iežaset ruovttubáiki.
 
 SELECT:r1699 Comp IF (-1 ("mihá"));
@@ -8107,7 +8107,7 @@ SELECT:r1836 VGen IF (0 VGENV LINK NOT 0
 	## Mánná viegai ovtta čieru ruoktot.
 	## Juoigga bođii ja garrut manai.
 
-SELECT:r1837 VGen IF (-1 COPULAS BARRIER NOT-ADV-PCLE)(0 VGENV LINK NOT 0 Der_Pass OR Loc OR Inf);  
+SELECT:r1837 VGen IF (-1 COPULAS)(0 VGENV LINK NOT 0 Der_Pass OR Loc OR Inf);  
 	## Muhtumat ledje vácci, ja muhtumat ledje čuoigga. 
 	
 SELECT:r1838 VGen IF (-2 COPULAS)(-1 Indef OR Adv)(0 VGENV LINK NOT 0 Der_Pass OR Loc OR Inf);  	
@@ -8429,8 +8429,8 @@ MAP:r95 (@→N) TARGET PrfPrc IF (*-1 Pr
 	## 1978 láhka balansere gaskal sámi vuoigatvuođaid boarásmuvvan oainnu,...
 
 
-MAP:r96 (@→N) TARGET PrfPrc IF (*-1 V-NOT-AUX LINK NOT 0 PrfPrc BARRIER S-BOUNDARY2 OR AUX LINK NOT 0 A OR N)(NEGATE *-1 COPULAS BARRIER NOT-QMARK)(0 Der_Pass)(*1 N BARRIER NOT-QMARK)(NEGATE *1 TIME BARRIER NOT-QMARK); 
-MAP:r97 (@→N) TARGET PrfPrc IF (*-1 S-BOUNDARY2 BARRIER NPNHA LINK NOT 0 A OR N)(0 Der_Pass)(*1 N LINK NOT 0 TIME BARRIER NPNHA);
+MAP:r96 (@→N) TARGET PrfPrc IF (*-1 (V-NOT-AUX LINK NOT 0 PrfPrc) BARRIER S-BOUNDARY2 OR AUX LINK NOT 0 A OR N)(NEGATE *-1 COPULAS BARRIER NOT-QMARK)(0 Der_Pass)(*1 N BARRIER NOT-QMARK)(NEGATE *1 TIME BARRIER NOT-QMARK); 
+MAP:r97 (@→N) TARGET PrfPrc IF (*-1 S-BOUNDARY2 BARRIER NPNHA LINK NOT 0 A OR N)(0 Der_Pass)(*1 (N LINK NOT 0 TIME) BARRIER NPNHA);
   	## Ráđđehus háliida lágidit dearvvašvuođa- ja sosiálabálvalusaid oktagaslaš dárbbuid mielde , earret eará pasientavuoigatvuođaid lágain ja dearvvašvuođabargiid lágain.
 
 MAP:r98 (@→N) TARGET PrfPrc IF (*-1 Adv BARRIER NOT-QMARK LINK *-1 COPULAS BARRIER NOT-ADV-PCLE LINK *-1 Nom BARRIER V-NOT-AUX OR S-BOUNDARY2) (*1C Nom BARRIER NPNHA LINK *1 NP-BOUNDARY BARRIER NOT-QMARK);
@@ -8643,7 +8643,7 @@ MAP (@+FMAINV) TARGET VFIN IF (0 ("leat"
 	## Nemi ja Pondus leaba sárgojuvvon hámit.
 	## Sus leat hervejuvvon gálssohat main heaŋgájit diehpit.
 
-MAP (@+FMAINV) TARGET VFIN IF (0 ("leat"))(1 TIME-ADV LINK 1 (N Nom) OR (A Nom) OR (Num Nom) BARRIER NPNHA LINK *-1 VFIN LINK -1 NP-HEAD-NOM OR HAB-ACTOR + Loc OR TIME OR TIME-MARKER OR DOPPE)(NEGATE *1 N OR A LINK *1 (Actio Ess) OR PrfPrc BARRIER VFIN OR SV-BOUNDARY);
+MAP (@+FMAINV) TARGET VFIN IF (0 ("leat"))(1 TIME-ADV LINK 1 (N Nom) OR (A Nom) OR (Num Nom) LINK *-1 VFIN LINK -1 NP-HEAD-NOM OR HAB-ACTOR + Loc OR TIME OR TIME-MARKER OR DOPPE)(NEGATE *1 N OR A LINK *1 (Actio Ess) OR PrfPrc BARRIER VFIN OR SV-BOUNDARY);
 	## Mearrasámiinge ledje ovdal moadde dábmojuvvon bohcco maid guođohedje, muhto dat jávke loahpas siseatnama sámiid bohccuid sisa. # ovdal er kan også være Pr, derfor fungerer ikke NPNHA som barrier
 
 
@@ -9047,7 +9047,7 @@ SELECT:r1992 Acc IF (0 NUMERALS)(1 MEASU
 SELECT:r1993 Gen IF (-1 (Num Sg Acc))(0 MEASURE)(1 ALU OR A);
 	## Min viessu lea vihtta mehtara alu.
 	## Sus lea njeallje mehtara guhkkosaš biila. 
-SELECT:r1994 Acc IF (0 NUMERALS BARRIER NOTCOMMA)(*1 (Num Acc) LINK 1 MEASURE LINK 1 ALU OR A);
+SELECT:r1994 Acc IF (0 NUMERALS)(*1 (Num Acc) LINK 1 MEASURE LINK 1 ALU OR A);
 	## Guokte golbma mehtera guhku, ja guhkitge, sáhttá vuojahat leat. 
 SELECT:r1995 Gen IF (1 (Num Sg Acc) LINK 1 (Num Sg Acc) LINK 1 MEASURE LINK 1 ALU OR A);
 
@@ -9293,7 +9293,7 @@ REMOVE:r2055 Gen IF (*-1 DURATION-A BARR
     ## Bohccot šaddet gárddis jorrat guhkes beaivvi 20-30 buolašgrádas.
  
 REMOVE:r2056 Gen IF (*0 PLACE-V OR DURATION-V BARRIER SV-BOUNDARY2)(0 VAHKKU-DUR + Acc LINK NOT 1 Ill OR TIME OR Po OR PREGEN LINK NEGATE 0 Nom LINK 1 EOS)(0 VAHKKU-DUR + Acc LINK NOT *-1 Dem OR TIME-A OR Ord OR Indef OR PREGEN OR Pr BARRIER NPNH);
-REMOVE:r2057 Gen IF (*-1 PLACE-V OR DURATION-V LINK 0 PrfPrc OR Inf BARRIER SV-BOUNDARY2 LINK *-1 AUX OR Neg BARRIER SV-BOUNDARY2)(0 VAHKKU-DUR + Acc LINK NOT *-1 Dem OR TIME-A OR Ord BARRIER NPNH LINK NEGATE *1 VAHKKU-DUR + Acc LINK 1 Po);
+REMOVE:r2057 Gen IF (*-1 PLACE-V OR DURATION-V LINK 0 PrfPrc OR Inf LINK *-1 AUX OR Neg BARRIER SV-BOUNDARY2)(0 VAHKKU-DUR + Acc LINK NOT *-1 Dem OR TIME-A OR Ord BARRIER NPNH LINK NEGATE *1 VAHKKU-DUR + Acc LINK 1 Po);
      ## Skiippaguras Kárášjohkii ádjánii bures beaivvi, muhto vuloseano ii ádjánan go beaivvi go dábálaš olu čáhci lei (Fixdal 1938: 64).
     
 #SELECT:r2058 Acc IF (*-1 PLACE-V OR DURATION-V OR ("leat") BARRIER SV-BOUNDARY2)(0 VAHKKU-DUR + Acc LINK 1 DOPPE OR Loc);
@@ -9307,7 +9307,7 @@ REMOVE:r2059 Acc IF (-1 POINT-IN-TIME-SP
 	
 REMOVE:r2060 Acc IF (*-1 POINT-IN-TIME-V BARRIER SV-BOUNDARY2)(0 TIME + Acc);
     ## Bivdu, eandálii beatnagiin bivdin, sáhttá hehttet boazobarggu čohkken- ja johtináiggi.
-REMOVE:r2061 Acc IF (*-1 POINT-IN-TIME-V LINK 0 PrfPrc OR Inf BARRIER SV-BOUNDARY2 LINK *-1 AUX OR Neg BARRIER SV-BOUNDARY2)(0 TIME + Acc);
+REMOVE:r2061 Acc IF (*-1 (POINT-IN-TIME-V LINK 0 PrfPrc) OR Inf BARRIER SV-BOUNDARY2 LINK *-1 AUX OR Neg BARRIER SV-BOUNDARY2)(0 TIME + Acc);
     # Can still be Acc @ADVL if standing after a DURATION-A
     # That is why these rules have to stand after the DURATION-A rule
 
@@ -9896,7 +9896,7 @@ REMOVE:r2198 (N Acc) IF (-1 Adv)(0 Gen)(
 REMOVE:r2199 (N Acc) IF (*-1C NP-BOUNDARY BARRIER WORD)(NEGATE -1 V)(NEGATE 1 Gen LINK 1 Po)(NEGATE 1 VFIN)(*1 N BARRIER NPNH OR CC LINK *1 VFIN BARRIER S-BOUNDARY2 OR CC)(NEGATE *1 NP-HEAD-NOM BARRIER NOT-NP OR CC LINK *1 V-TRANS-ACT BARRIER SV-BOUNDARY2 OR CC LINK NOT *1 NP-HEAD-ACC BARRIER S-BOUNDARY2 OR CC)(NEGATE 0 ("hearrá" Acc) LINK 1 ("ipmil" Acc))(NEGATE 1 Ill LINK *1 VFIN BARRIER NOT-ADV-PCLE);
 	## Ášši meannudeapmi loahpahuvvui guovvamánu 20. b. dii. 09.15. 
 
-REMOVE:r2200 (N Nom) IF (*-1 NP-BOUNDARY BARRIER WORD)(NEGATE -1 COPULAS)(0 Gen)(NEGATE 0 TITLE)(NEGATE 1 COPULAS OR Inf)(*1 (N Nom) BARRIER NOT-ADJ OR CLB OR ("/") LINK *1 VFIN BARRIER S-BOUNDARY2 OR CC OR PUNCT)(NEGATE 1 VFIN BARRIER NOT-ADV-PCLE);
+REMOVE:r2200 (N Nom) IF (*-1 NP-BOUNDARY BARRIER WORD)(NEGATE -1 COPULAS)(0 Gen)(NEGATE 0 TITLE)(NEGATE 1 COPULAS OR Inf)(*1 (N Nom) BARRIER NOT-ADJ OR CLB OR ("/") LINK *1 VFIN OR CC OR PUNCT)(NEGATE 1 VFIN);
 	## Ášši meannudeapmi loahpahuvvui guovvamánu 20. b. dii. 09.15. 
 	## Maiddái álgoálbmotfestivála Riddu Riđđu doarjja unnui.
  
@@ -10017,7 +10017,7 @@ SELECT:r2235 Gen IF (*-1 ("dego") OR ("d
 REMOVE:r2236 Acc IF (-2 ADVLCASE)(-1 go)(*1 ADVLCASE BARRIER NPNH);
 	## Dat gullá lunddolaččat dakkár áššiide go Sámedikkiid válganjuolggadusaide.
 
-REMOVE:r2237 Acc IF (-1 ("sárdnut") BARRIER NOT-ADV-PCLE)(0 Gen)(*1 ("birra") BARRIER NPNH);
+REMOVE:r2237 Acc IF (-1 ("sárdnut"))(0 Gen)(*1 ("birra") BARRIER NPNH);
 	## Mii gullat sin sárdnumin Ipmila stuora daguid birra min iežamet gillii.
 
 REMOVE:r2238 Acc IF (*-1 ("birra") BARRIER NPNHA LINK *-1 ("sárdnut") BARRIER SV-BOUNDARY2 OR CC);
@@ -10388,7 +10388,7 @@ SELECT:r2331 (N Nom) IF (*-1 (Dem Nom) B
 SELECT:r2332 (Num Nom) IF (-1 (Dem Sg Nom));
 	## Dát guokte ášši Siviilaáittardeaddji geavadis leat váldon...
 	
-SELECT:r2333 (Num Nom) IF (1 TIME + Gen BARRIER NOT-ADJ LINK *1 ("leat") BARRIER N LINK *1 Nom BARRIER NPNH) ;
+SELECT:r2333 (Num Nom) IF (1 TIME + Gen LINK *1 ("leat") BARRIER N LINK *1 Nom BARRIER NPNH) ;
 	## Vuosttaš logi minuvtta lei buorre áigodat Nordlysa ektui.
 
 SELECT:r2334 Nom IF (*-1 BOC2 OR PUNCT BARRIER NPNHA OR (Dem Acc))(0 Sg OR ABBR LINK NOT 0 Ord)(NEGATE 0 Adv OR Loc OR Po OR VFIN)(NEGATE 0 Ess LINK *1 Nom BARRIER NPNH)(NEGATE 0 Attr LINK *1 N BARRIER NPNH)(*1C V-SG3 BARRIER NOT-ADV-PCLE);
@@ -10628,7 +10628,7 @@ SELECT:r2402 Nom IF (-1 CRD LINK *-1C (N
 SELECT:r2403 Nom IF (-2C Nom)(-1 CRD)(1 Gen LINK *1 Po BARRIER NPNH)(NEGATE *1 Nom BARRIER Po);
 	## Dearvvašvuođaásahusas mearriduvvo divvun dahje sihkkun dearvvašvuođabargiidlága § 44 mielde.
 	
-SELECT:r3529 Nom IF (*-1 TITLE LINK 0 ADVLCASE BARRIER NPNH)(0 Prop);
+SELECT:r3529 Nom IF (*-1 (TITLE LINK 0 ADVLCASE) BARRIER NPNH)(0 Prop);
 	## Son logai dan maiddái professoriidda Asbjørn Kjønstad ja Kristian Andenæs.	
 		## Son logai dan maiddái professoriidda Kjønstad ja Andenæs.	
 
@@ -10728,7 +10728,7 @@ SELECT:r2429 Loc IF (*-1 HEADOFPARTS BAR
 	## Guhtemuš dainna bivdovugiin guhkimus leaš gevahuvvon...
 	## Dat sáhttet leat oassin árvvoštallamiin. #If this does not work: maybe:REMOVE Acc?
 	
-SELECT:r2430 Loc IF (*-1 HEADOFPARTS LINK 1 Gen BARRIER NPNH OR COMMA);
+SELECT:r2430 Loc IF (*-1 (HEADOFPARTS LINK 1 Gen) BARRIER NPNH OR COMMA);
   # Just in case there is a genitive which still is an accusative as well and therefore NPNH and ruled out by the rule before.
 
 	# Accusative
@@ -10851,7 +10851,7 @@ SELECT:r2455 Acc IF (*-1 VFIN BARRIER NP
 
 	# In front of a participle
 	# - - - - - - - - - - - - -
-SELECT:r2456 Acc IF (*-1 LEAT-FIN-NOT-IMP BARRIER NPNHAI)(NEGATE 0 Adv OR Nom OR TIME)(*1 PrfPrc LINK 0 V-TRANS-ACT BARRIER NOT-ADV LINK NOT *1 ("leat" PrfPrc) BARRIER S-BOUNDARY2);
+SELECT:r2456 Acc IF (*-1 LEAT-FIN-NOT-IMP BARRIER NPNHAI)(NEGATE 0 Adv OR Nom OR TIME)(*1 (PrfPrc LINK 0 V-TRANS-ACT) BARRIER NOT-ADV LINK NOT *1 ("leat" PrfPrc) BARRIER S-BOUNDARY2);
 	## Leago barggu ohcan?
 
 SELECT:r2458 Acc IF (1 ("jahki"))(2 ("deavdit"));
@@ -10897,7 +10897,7 @@ SELECT:r2466 Acc IF (*-1C V-TRANS-ACT BA
 	## Sámegillii galbet universitehta lea šaddan realitehta.
 
 
-SELECT:r2467 Acc IF (*-1 Loc BARRIER NPNH LINK -1C V-TRANS-ACT BARRIER SV-BOUNDARY2 OR NPNHA OR ("nugo") OR ("dego") OR ("dugo") LINK NOT 0 FAUXV LINK NOT 0 Gen)(NEGATE 0 (Sg Nom) LINK *-1 (V Sg3) BARRIER SV-BOUNDARY2)(NEGATE -1 PrfPrc LINK NOT *-1 REALCOPULAS BARRIER SV-BOUNDARY2)(NEGATE -1 NUMERALS)(NEGATE 0 PrfPrc OR Adv OR VFIN OR Loc OR TIME)(1C NP-BOUNDARY LINK NOT 0 Actio);
+SELECT:r2467 Acc IF (*-1 Loc BARRIER NPNH LINK -1C V-TRANS-ACT OR NPNHA OR ("nugo") OR ("dego") OR ("dugo") LINK NOT 0 FAUXV LINK NOT 0 Gen)(NEGATE 0 (Sg Nom) LINK *-1 (V Sg3) BARRIER SV-BOUNDARY2)(NEGATE -1 PrfPrc LINK NOT *-1 REALCOPULAS BARRIER SV-BOUNDARY2)(NEGATE -1 NUMERALS)(NEGATE 0 PrfPrc OR Adv OR VFIN OR Loc OR TIME)(1C NP-BOUNDARY LINK NOT 0 Actio);
 	## Dat geahččá mus borramuša.
 
 SELECT:r2468 Acc IF (*-1C V-TRANS-ACT-NOT-ACT BARRIER COMMA OR Num OR Acc OR Nom OR VERB OR Ess LINK 0 FMAINV OR Inf LINK NOT *-1 Acc BARRIER S-BOUNDARY2)(NEGATE 0 Adv OR TIME)(NEGATE 0 Gen LINK *1 Acc OR Ess BARRIER VERB OR S-BOUNDARY2)(NEGATE 0 ("nubbi") LINK 1 A)(NEGATE 0 ANIMATE LINK NOT *1 ANIMATE BARRIER NPNH)(NEGATE 0 Gen LINK -1 Pr)(NEGATE 0 Px LINK 0 Loc) ;
@@ -11006,7 +11006,7 @@ SELECT:r2492 Acc IF (*-1 V-TRANS-ACT BAR
 	## Mun oidnen Máreha boahtime.
  # The rule also hits objects in front of Actio Ess, which seems OK.
  
-REMOVE:r2493 Acc IF (*-1 ("bargat") LINK 0 FMAINV OR Inf BARRIER NOT-ADV-PCLE)(0 Gen)(1 (N Ess));
+REMOVE:r2493 Acc IF (*-1 ("bargat") LINK 0 FMAINV OR Inf)(0 Gen)(1 (N Ess));
 	## Son lea bargan áviissa váldodoaimmaheaddjin.
 
 REMOVE:r2494 Acc IF (*-1 Acc BARRIER NOT-NP OR Inf OR CLB LINK *-1 OPREDV BARRIER NPNH)(NEGATE *-1 ("/") OR CC BARRIER NPNH LINK -1 Acc)(1 Gen LINK *1 Po BARRIER NPNH);
@@ -11109,7 +11109,7 @@ SELECT:r2516 Acc IF (*-1C V-TRANS-ACT BA
 SELECT:r2517 Acc IF (1 ("/") LINK *1 NP-HEAD-ACC BARRIER NPNH LINK 1 NP-BOUNDARY);
 	## Lea dárbu ásahit sámegiela vierisgiellaoahpahusa/álgooahpahusa.
  
-SELECT:r2518 Acc IF (*-1 V-TRANS-ACT BARRIER NPNHAII)(0 N)(1 COMMA)(*2 NP-HEAD-ACC BARRIER NPNHA LINK 1 CC LINK 1 NP-HEAD-ACC BARRIER NPNHA LINK 1 NP-BOUNDARY);
+SELECT:r2518 Acc IF (*-1 V-TRANS-ACT BARRIER NPNHAII)(0 N)(1 COMMA)(*2 NP-HEAD-ACC BARRIER NPNHA LINK 1 CC LINK 1 NP-HEAD-ACC LINK 1 NP-BOUNDARY);
      	## Jos juohká dážaid, sámiid ja láddelaččaid iešguđet sadjái, ...
 
 SELECT:r2520 Acc IF (*-1 COMMA BARRIER NPNH LINK -1C NP-HEAD-ACC)(1 COMMA OR CC)(2C Gen)(3C Acc);
@@ -11142,7 +11142,7 @@ SELECT:r2528 Acc IF (*-1 (@CNP) BARRIER
 
 	# Third part
 	# - - - - - -
-SELECT:r2529 Acc IF (*-1 CC BARRIER NPNH LINK -1C Acc LINK -1 COMMA BARRIER NPNH LINK -1C Acc)(1 NP-BOUNDARY OR Gen);
+SELECT:r2529 Acc IF (*-1 CC BARRIER NPNH LINK -1C Acc LINK -1 COMMA LINK -1C Acc)(1 NP-BOUNDARY OR Gen);
 	## Ferte vuhtiiváldit divššohasa dárbbuid, beroštumiid ja riektesihkarvuođa dearvvašvuođabálvalusa ektui. 
 
 
@@ -11253,7 +11253,7 @@ SELECT:r2556 Ill IF (*-1 CC BARRIER NPNH
 
 	# Early plural rule
 	# -----------------
-SELECT:r2557 Pl IF (0 SG-WORD)(-1 (Pron Gen) LINK -1 Num OR Ord BARRIER NPNH);
+SELECT:r2557 Pl IF (0 SG-WORD)(-1 (Pron Gen) LINK -1 Num OR Ord);
    	## Dát lea okta min ovttasbargguin.
     # This rule is necessary before the next one as it chooses plural even if the genitive pronoun modifier after the numeral and in front of the singular noun is not disambiguated yet (Acc-Gen disambiguation comes later).
 
@@ -11313,7 +11313,7 @@ SELECT:r2579 Com IF (0 (Pl Loc))(*1 COM-
 	
 SELECT:r2580 Com IF (*-1 COM-ADV BARRIER (N Com));
 	## Ii lean dárbu dábálaš eanavuovdimii, oktan opmodatregistreremiin. 
-SELECT:r2581 Com IF (*-1 ("guovttá"))(0 Loc LINK 0 ANIMATE BARRIER S-BOUNDARY2);
+SELECT:r2581 Com IF (*-1 ("guovttá"))(0 Loc LINK 0 ANIMATE);
   	## Dat son muitalii, go guovttá leimme dainna nieiddain.
 
 SELECT:r2582 Com IF (*-1 ("váldit") BARRIER NPNH LINK -1 ("oktavuohta"));
@@ -11367,7 +11367,7 @@ SELECT:r2597 Com IF (*-1C (Indef Sg Com)
     ## Dainna sániin dadjat, de mearkkaša dat ahte ...
 
 SELECT:r2599 Com IF (0 Num)(NEGATE *1 PL-NOUN BARRIER NOT-ADJ LINK 0 Pl); # Cf. rules for PL-NOUN above.
-SELECT:r2600 Com IF (*-1C Num LINK 0C Com BARRIER NOT-ADJ OR ("jahkásaš"));
+SELECT:r2600 Com IF (*-1C (Num LINK 0C Com) BARRIER NOT-ADJ OR ("jahkásaš"));
 	## ...de son gal luoitá nođiid; ovttain sániin dadjat: luoitala.
 	## Mo sáhttá mottiin sániin dadjat...
 
@@ -11481,7 +11481,7 @@ SELECT:r2619 Loc IF (-3 IV)(-2 Nom)(-1 E
 	# Words which should have Loc, we put into SG-WORD, like:
 	## Ii oppa Salomo ge buot hearvásvuođainis lean gárvvohuvvon nugo okta sis. 
 	
-SELECT:r2621 (Pl Loc) IF (*-1 ("dievva"))(0 (Sg Com) BARRIER NPNH);
+SELECT:r2621 (Pl Loc) IF (*-1 ("dievva"))(0 (Sg Com));
   	## Dušše diibmabeali ovdalaš lei birasviessu dievva Sámi nuorain.
   	
 
@@ -11685,7 +11685,7 @@ SELECT:r2680 Com IF (*-1 COMV BARRIER SV
 SELECT:r2681 Com IF (0 Sg)(*1 COMV BARRIER SV-BOUNDARY2 LINK 0 VERB)(NEGATE *1 COPULAS BARRIER VERB LINK *1 COMV BARRIER NOT-ADV-PCLE LINK 0 Inf);
   	## ...go sápmelaččat duođas álge eanadoaluin bargat.
 
-REMOVE:r2682 (Pl Loc) IF (*-2 OKTII-V OR COMV)(*-1 ("oktii"))(0 N BARRIER Loc);
+REMOVE:r2682 (Pl Loc) IF (*-2 OKTII-V OR COMV)(*-1 ("oktii"))(0 N);
   	## Mii heive bures oktii ránes datneláiggiin.
   # we have to think about a barrier, another Loc to the right for example
   
@@ -11769,14 +11769,14 @@ SELECT:r2700 Loc IF (0 HUMAN OR ORGANIZA
 	## Guovlluid dearvvašvuođafitnodagain gáibiduvvojit dakkár rutiinnat.
 
 SELECT:r2701 Loc IF (*-1 SOURCE-V BARRIER SV-BOUNDARY2 LINK NOT 0 Adv)(NEGATE *-1 CC BARRIER S-BOUNDARY2 LINK -1 LOCV)(0 HUMAN OR ORGANIZATION); 
-SELECT:r2702 (Pl Loc) IF (*-1 NP-HEAD-NOM LINK 0 HUMAN OR ORGANIZATION OR (Pron Rel))(0 ORGANIZATION OR HUMAN OR EVENT LINK 0 (Sg Com) LINK NOT -1 PRON-DU)(*1 HUMAN-AGENT-V LINK NOT 0 Actio LINK NOT 1 N BARRIER S-BOUNDARY2 OR COMMA);
+SELECT:r2702 (Pl Loc) IF (*-1 NP-HEAD-NOM LINK 0 HUMAN OR ORGANIZATION OR (Pron Rel))(0 ORGANIZATION OR HUMAN OR EVENT LINK 0 (Sg Com) LINK NOT -1 PRON-DU)(*1 (HUMAN-AGENT-V LINK NOT 0 Actio LINK NOT 1 N) BARRIER S-BOUNDARY2 OR COMMA);
   	## Gielddat berrejit dearvvašvuođastašuvnnain deattuhit mánáid ja nuoraid váttisvuođaid ovddalgihtii eastadeami ja psykososiálabarggu mánáid ja nuoraid váras.
 
-SELECT:r2703 (Pl Loc) IF (*-1 HUMAN-AGENT-V LINK NOT 0 Actio LINK NOT 1 N)(0 ORGANIZATION OR EVENT LINK 0 (Sg Com) BARRIER S-BOUNDARY2 OR COMMA LINK NOT -1 PRON-DU);
+SELECT:r2703 (Pl Loc) IF (*-1 HUMAN-AGENT-V LINK NOT 0 Actio LINK NOT 1 N)(0 ORGANIZATION OR EVENT LINK 0 (Sg Com) OR COMMA LINK NOT -1 PRON-DU);
   	## Departemeanta áigu ovttasbargat Sámedikkiin ja movttiidahttit gielddaid ovddidit doaimmaid, mat nannejit sámi mánáid ja nuoraid identitehta ja oktiigullevašvuođa sámi servodagain.
   # This rules assumes that abstract activities are performed with a tool but not with the help of others in comitative case. The set of abstract-activity verbs might have to be restricted to those verbs that do have these features.
   
-SELECT:r2704 (Pl Loc) IF (0 ORGANIZATION OR HUMAN OR EVENT LINK 0 (Sg Com) BARRIER S-BOUNDARY2 OR COMMA LINK NOT -1 PRON-DU)(*1 HUMAN-AGENT-V LINK NOT 0 Actio LINK NOT 1 N);
+SELECT:r2704 (Pl Loc) IF (0 ORGANIZATION OR HUMAN OR EVENT LINK 0 (Sg Com) OR COMMA LINK NOT -1 PRON-DU)(*1 HUMAN-AGENT-V LINK NOT 0 Actio LINK NOT 1 N);
   
 SELECT:r2705 (N Pl) IF (0 (Sg Com))(1C (Pron Rel Pl));
   	## Áigodatbajilgovva nissoniin geat leat válljejuvvon Sámediggái 1989-2001 54 Tab.4 .
@@ -11874,7 +11874,7 @@ SELECT:r2725 (Sg Com) IF (*0 VERBAL-ACTI
 SELECT:r2727 (Pl Loc) IF (*-1 VERBAL-ACTIVITY-V BARRIER SV-BOUNDARY2 LINK 0 FMAINV OR Inf)(0 Com LINK 0 EVENT OR EVENT-TOOL-ACTIVITY);
   	## Dan geažil lea dehálaš ávžžuhit bargiid sámástit ráđđádallamiin.
   
-SELECT:r2728 (Pl Loc) IF (NEGATE *-1 COM-HEAD BARRIER NPNH)(0 TEXT OR ABSTR-TEXT)(*0 VERBAL-ACTIVITY-V LINK 0 Der_Pass BARRIER NPNH);
+SELECT:r2728 (Pl Loc) IF (NEGATE *-1 COM-HEAD BARRIER NPNH)(0 TEXT OR ABSTR-TEXT)(*0 (VERBAL-ACTIVITY-V LINK 0 Der_Pass) BARRIER NPNH);
   	## Dálá sámiid máidnasiin ja muitalusain muitaluvvo dološ oskui gullan hálddiid birra .
   	## Sii máinnašuvvojedje vearrologahallamiin earáid maŋŋá . 
   	
@@ -11948,13 +11948,13 @@ SELECT:r2750 (Sg Com) IF (0 (Pl Loc) LIN
 
 # HUMAN-LOC-V 
 # - - - - - - 
-SELECT:r2752 (Pl Loc) IF (0 (Sg Com) LINK 0 SATNI)(*1 HUMAN-LOC-V BARRIER SV-BOUNDARY2 LINK 0 FMAINV OR Inf LINK *1 WORD-PARTS OR SINGLE-SOUND LINK 0 Acc OR Nom BARRIER SV-BOUNDARY2);
+SELECT:r2752 (Pl Loc) IF (0 (Sg Com) LINK 0 SATNI)(*1 HUMAN-LOC-V BARRIER SV-BOUNDARY2 LINK 0 FMAINV OR Inf LINK *1 WORD-PARTS OR SINGLE-SOUND LINK 0 Acc OR Nom);
     ## Váldonjuolggadus lea ahte báikenamain dábálaččat eat ane goallossázu.
     #(Act+Acc) WORD-PARTS becomes the THEME/PATIENT
     ## Movt báikenamain geavahuvvo goallossáhcu čilgejuvvo 2.1.5 čuoggás. 
     #(Pass+Nom)
 
-SELECT:r2753 (Pl Loc) IF (*0 HUMAN-LOC-V BARRIER SV-BOUNDARY2 LINK 0 FMAINV OR Inf LINK *1 WORD-PARTS OR SINGLE-SOUND LINK 0 Acc OR Nom BARRIER SV-BOUNDARY2)(0 (Sg Com) LINK 0 CONCEPT OR HUMAN OR SATNI);
+SELECT:r2753 (Pl Loc) IF (*0 HUMAN-LOC-V BARRIER SV-BOUNDARY2 LINK 0 FMAINV OR Inf LINK *1 WORD-PARTS OR SINGLE-SOUND LINK 0 Acc OR Nom)(0 (Sg Com) LINK 0 CONCEPT OR HUMAN OR SATNI);
     ## Atnit stuora bustáva lágaid namain:
     ## Gravismearka adnojuvvo sámegielas dušše muhtun vierronamain.
     # Have to preceede the next Sg Com rules because they are an exception to it with respect to SATNI
@@ -11971,7 +11971,7 @@ SELECT:r2755 (Sg Com) IF (0 (Pl Loc) LIN
 	# COPULAS 
 	# - - - - 
 SELECT:r2760 (Sg Com) IF (*-1 COPULAS BARRIER SV-BOUNDARY2 LINK 0 PrfPrc OR Inf OR ConNeg OR VFIN)(0 (Pl Loc) LINK 0 ANIMATE OR CONCEPT LINK *-1 A BARRIER NP-BOUNDARY LINK NOT 0 Ord);
-SELECT:r2761 (Sg Com) IF (0 (Pl Loc) LINK 0 ANIMATE OR CONCEPT LINK *-1 A BARRIER NP-BOUNDARY)(*1 COPULAS LINK 0 Inf OR ConNeg OR VFIN BARRIER S-BOUNDARY2 LINK NEGATE *1 VERB LINK NEGATE 0 VFIN LINK NOT 0 COPULAS);
+SELECT:r2761 (Sg Com) IF (0 (Pl Loc) LINK 0 ANIMATE OR CONCEPT LINK *-1 A BARRIER NP-BOUNDARY)(*1 COPULAS LINK 0 Inf OR ConNeg OR VFIN LINK NEGATE *1 VERB LINK NEGATE 0 VFIN LINK NOT 0 COPULAS);
   	## Go dii ehpet leat šiega leamaš ealibiin.
   	## Go dii lehpet šiega leamaš ealibiin.
   	## Go dii lehpet šiega ealibiin.
@@ -12060,14 +12060,14 @@ SELECT:r2776 (Sg Loc) IF (*-1 CRD2 BARRI
 	## Go 110a:s lea sáhka sápmelaččaid "ealáhusaid" sihkkarastimis ja ovdánahttimis,...
 	## Don heahpanaddat viesus ja sániin dán oskkáldasmeahttun ja suttolaš sohkagottis, sus heahpana Olbmobárdni ge.
 
-SELECT:r2777 (Pl Loc) IF (*-1 CRD2 BARRIER NPNH LINK *-1 Loc LINK NOT 0 Com BARRIER NPNH);
+SELECT:r2777 (Pl Loc) IF (*-1 CRD2 BARRIER NPNH LINK *-1 (Loc LINK NOT 0 Com) BARRIER NPNH);
 	## Sámi oahpahussuorgi stivrejuvvo daid rámmaid mielde maid lágat ja njuolggadusat bidjet, ja daid ulbmiliid ja višuvnnaid mielde mat ovdanbohtet Sámediggeplánas ja eará oahpahus- ja oahppoplánain ja strategiijain.
 	## Mii maid muđui fuomášuhttit ahte ollu lohkoávdnasiin mat fátmmastit dahje čilgejit sámediggeválgga , dain lea sáhka smávva loguin / vuollegis lohkoárvvuin.
 
 SELECT:r2778 (Pl Loc) IF (1 CRD2)(*2 Loc BARRIER NPNH LINK NOT 0 Com);
 	## Son lea ođđa mediain ja skuvllas.
 	## Vuđolaš váttisvuohta lea dattege ahte sámi bustávat eai doaimma dohkálaččat ovttaskas dihtoriin eaige interneahtas. 	
-SELECT:r2779 Loc IF (*-1C Loc LINK 1 CRD2)(0 (Sg Com) BARRIER NPNH)(1 CRD2 LINK *1C Loc BARRIER NPNH);
+SELECT:r2779 Loc IF (*-1C Loc LINK 1 CRD2)(0 (Sg Com))(1 CRD2 LINK *1C Loc BARRIER NPNH);
 	## Dearvvašvuođaásahusain, doaktáriin, bátnedoaktáriin ja sealgeetniin lea leamaš geatnegasvuohta čállit journálaid maiddái ovddeš lágaid mielde. 
 SELECT:r2780 Loc IF (*-1C Loc LINK *1 Rel BARRIER Com OR S-BOUNDARY2)(0 Com LINK *-1 CRD2 BARRIER NPNH);
 
@@ -12108,9 +12108,9 @@ SELECT:r2789 Loc IF (NEGATE *-1 BOS LINK
 	# - - - - - - - - - - - - 
 
 	#The following rules are for two numerlas seperated by sth. that still makes sure we are talking about one NP:
-SELECT:r2790 Loc IF (NEGATE *-1 COM-HEAD BARRIER S-BOUNDARY2)(*-1 Num BARRIER NPNHA LINK -1 PRE-NP-HEAD OR Adv LINK -1 Num)(0 Com BARRIER NPNH)(NEGATE *-1 COM-HEAD BARRIER S-BOUNDARY2);
+SELECT:r2790 Loc IF (NEGATE *-1 COM-HEAD BARRIER S-BOUNDARY2)(*-1 Num BARRIER NPNHA LINK -1 PRE-NP-HEAD OR Adv LINK -1 Num)(0 Com)(NEGATE *-1 COM-HEAD BARRIER S-BOUNDARY2);
   	## Muhtin biirres leat gávcci obbalaččat 12 válljejuvvon áirasiin leamaš nissonolbmot, ja eará biirres leat buohkat geat leat válljejuvvon leamaš dievddut.
-SELECT:r2791 Loc IF (NEGATE *-1 COM-HEAD BARRIER S-BOUNDARY2)(-1C (PrfPrc @→N) LINK *-1 Num BARRIER NPNHA LINK -1 PRE-NP-HEAD OR Adv LINK -1 Num)(0 Com BARRIER NPNH)(NEGATE *-1 COM-HEAD BARRIER S-BOUNDARY2);
+SELECT:r2791 Loc IF (NEGATE *-1 COM-HEAD BARRIER S-BOUNDARY2)(-1C (PrfPrc @→N) LINK *-1 Num BARRIER NPNHA LINK -1 PRE-NP-HEAD OR Adv LINK -1 Num)(0 Com)(NEGATE *-1 COM-HEAD BARRIER S-BOUNDARY2);
 
 	#The following ones are for two numerals next to each other, here we have to make sure that those are not of the form "2-3 Nom:ain" written as "2 3 Nom:ain",:
 SELECT:r2792 Loc IF (NEGATE *-1 COM-HEAD BARRIER S-BOUNDARY2)(*-1 Num BARRIER NPNHA LINK 0 Sg OR Pl LINK -1 Num LINK NOT 0 Sg OR Pl)(0 Com)(NEGATE *-1 COM-HEAD BARRIER S-BOUNDARY2);
@@ -12140,7 +12140,7 @@ SELECT:r2797 Loc IF (*-1 N BARRIER NPNHA
 SELECT:r2798 Loc IF (*-1 ("ollu") OR ("olu") BARRIER NPNH OR COMMA);
 	## Ollu dainna bivdovugiin guhkimus leaš gevahuvvon...
 	
-SELECT:r2799 Loc IF (*-1 ("ollu") OR ("olu") LINK 1 Gen BARRIER NPNH OR COMMA);
+SELECT:r2799 Loc IF (*-1 ("ollu") OR ("olu") LINK 1 Gen OR COMMA);
   # Just in case there is a genitive which still is an accusative as well and therefore NPNH and ruled out by the rule before.
 
 
@@ -12764,7 +12764,7 @@ SELECT:r2974 V-PL3 IF (*-1 (Pl Nom) BARR
 
 SELECT:r2975 V-PL3 IF (*-1C NP-HEAD-PL-NOM BARRIER PPRON-NOT-SII OR (N Sg Nom) OR CS OR CC)(NEGATE *-1 (V Pl3) BARRIER SV-BOUNDARY2 OR PUNCT);
 	## Makkár vuoigatvuođat dain leat riikavuložin? 
-SELECT:r2976 V-PL3 IF (1C NP-HEAD-PL-NOM BARRIER PPRON-NOT-SII);
+SELECT:r2976 V-PL3 IF (1C NP-HEAD-PL-NOM);
 
 SELECT:r2977 V-PL3 IF (*-1C (N Pl Nom) BARRIER MII-PERS OR MOAI OR Nom OR SV-BOUNDARY2);
 	## Dárbbašlaš dieđut dakkár mediaid vuođul sáhttet čállojuvvot journálii. 
@@ -12803,9 +12803,9 @@ SELECT:r2985 V-PL3 IF (*-1 VFIN BARRIER
 
  # Pl3 verbs that follow appositions:
  
-REMOVE:r2986 (V Sg2) IF (*-1 NUMERALS + Nom BARRIER VFIN LINK 1 (N Gen) BARRIER ("don"));
+REMOVE:r2986 (V Sg2) IF (*-1 NUMERALS + Nom BARRIER VFIN LINK 1 (N Gen));
 
-REMOVE:r2987 (V Pl1) IF (*-1 NUMERALS + Nom BARRIER VFIN LINK 1 (N Gen) BARRIER ("mii"));
+REMOVE:r2987 (V Pl1) IF (*-1 NUMERALS + Nom BARRIER VFIN LINK 1 (N Gen));
   	## Guokte beroštumiorganisašuvnna, Kirkens Familievern (KF, Girku bearašsuodjalus) ja Offentlige Familievernkontorers Organisasjon (OFO, Almmolaš bearašsuodjaluskantuvrraid organisašuvdna) leat dán rádjai fuolahan doaimmain ja prošeavttain, maid ulbmil lea ovddidit fágalaš bearašsuodjalusa.
 
  # Coordinated subjects
@@ -12969,7 +12969,7 @@ SELECT:r3028 Inf IF (-1 ("mo") OR ("mot"
  # There are two rules in the beginning, conflate when I see the effect.
  # The barrier may be too weak.
 
-SELECT:r3029 Inf IF (-1 A BARRIER SV-BOUNDARY2 LINK *-1 REALCOPULAS OR ("šaddat") BARRIER NOT-ADV)(NEGATE 0 (N Pl Nom));
+SELECT:r3029 Inf IF (-1 A LINK *-1 REALCOPULAS OR ("šaddat") BARRIER NOT-ADV)(NEGATE 0 (N Pl Nom));
 	## Eará guovlluin lei vejolaš nannet dárogiela.
 
 SELECT:r3030 Inf IF (*-1 N BARRIER SV-BOUNDARY2 LINK *-1 REALCOPULAS OR ("šaddat") BARRIER NPNHAI)(NEGATE 0 (N Pl Nom));
@@ -13319,8 +13319,8 @@ REMOVE:r3130 ("eallát") IF (0 ("eallit"
 	# gerget takes Acc     
 
 REMOVE:r3133 ("gerget") IF (0 ("geargat"))(NEGATE *1 Acc BARRIER NPNH);
-REMOVE:r3134 ("gerget") IF (0 ("geargat"))(*1 Acc LINK 1 Inf BARRIER NPNH);
-REMOVE:r3135 ("gerget") IF (0 ("geargat"))(*1 Acc LINK -1 Inf BARRIER NPNH);
+REMOVE:r3134 ("gerget") IF (0 ("geargat"))(*1 (Acc LINK 1 Inf) BARRIER NPNH);
+REMOVE:r3135 ("gerget") IF (0 ("geargat"))(*1 (Acc LINK -1 Inf) BARRIER NPNH);
 
 REMOVE:r3136 ("geavvat") IF (0 ("geavahit"));
 
@@ -13928,11 +13928,11 @@ MAP:r46 (@→N) TARGET (Pron Attr)(*1 N
 MAP:r47 (@N←) TARGET ("eatnat" Indef) IF (*-1 ("giittos") BARRIER NOT-QMARK);
 	## Giittos eatnat luoikamis.
 
-MAP:r48 (@Pron←) TARGET ("goappašagat" Nom) IF (*-1 PPRON-DU-PL LINK 0 Nom BARRIER S-BOUNDARY2)(NEGATE *-1 CC OR CLB BARRIER NOT-QMARK)(NEGATE *1 Rel BARRIER NOT-QMARK);
+MAP:r48 (@Pron←) TARGET ("goappašagat" Nom) IF (*-1 (PPRON-DU-PL LINK 0 Nom) BARRIER S-BOUNDARY2)(NEGATE *-1 CC OR CLB BARRIER NOT-QMARK)(NEGATE *1 Rel BARRIER NOT-QMARK);
 MAP (@Pron←) TARGET ("goappašagat" Loc) (*-1 DUPRON + Loc BARRIER S-BOUNDARY2)(NEGATE -1 CC OR CLB )(NEGATE 1 Rel);
 	## Soai leaba goappašagat nu jallat.
 
-MAP:r49 (@N←) TARGET ("goappašagat" Nom) IF (*-1 (N Pl) LINK 0 Nom BARRIER S-BOUNDARY2)(NEGATE *-1 CC OR CLB BARRIER NOT-QMARK)(NEGATE *1 Rel BARRIER NOT-QMARK);
+MAP:r49 (@N←) TARGET ("goappašagat" Nom) IF (*-1 (N Pl) LINK 0 Nom)(NEGATE *-1 CC OR CLB BARRIER NOT-QMARK)(NEGATE *1 Rel BARRIER NOT-QMARK);
 
 MAP:r526 (@←SUBJ) TARGET ("ieš" Refl Sg Nom) IF (1 (Refl PxSg3));
 	## Sámi ássiin lea nana árbevierru veahkehit ieš iežas persovnnalaš fierpmádagaid vehkiin.
@@ -14250,7 +14250,7 @@ MAP (@→Num) TARGET Num IF (1 NUMUNIT L
  
 MAP:r154 (@→N) TARGET (Num Acc) IF (*1 MEASURE BARRIER NOT-QMARK LINK *1 ALU BARRIER NOT-QMARK);
 	## Guokte golbma mehtera guhku, ja guhkitge, sáhttá vuojahat leat. 
-MAP:r155 (@→N) TARGET (Num Acc) IF (0 Num BARRIER NOTCOMMA)(*1 (Num Acc) LINK *1 MEASURE BARRIER NOT-QMARK LINK *1 ALU BARRIER NOT-QMARK);
+MAP:r155 (@→N) TARGET (Num Acc) IF (0 Num)(*1 (Num Acc) LINK *1 MEASURE BARRIER NOT-QMARK LINK *1 ALU BARRIER NOT-QMARK);
 	## Guokte golbma mehtera guhku, ja guhkitge, sáhttá vuojahat leat. 
 MAP:r156 (@→N) TARGET (Num Acc) IF (*1 (Num Acc) BARRIER NOT-QMARK LINK *1 (Num Acc) BARRIER NOT-QMARK LINK *1 MEASURE BARRIER NOT-QMARK LINK *1 ALU BARRIER NOT-QMARK);
 	## Guokte golbma njeallje mehtera guhku, ja guhkitge, sáhttá vuojahat leat.
@@ -14317,7 +14317,7 @@ MAP:r171 (@Pron←) TARGET (Coll Nom) IF
 #MAP:r172 (@←SUBJ @SUBJ→ @SPRED) TARGET (Coll Nom);
 	## Go sotnabeaivi šattai, de čuoigaledje máŋggas várrái. 
 
-MAP:r173 (@SUBJ→) TARGET (Num Nom) IF (1 TIME + Gen BARRIER NOT-ADJ LINK *1 ("leat") BARRIER N LINK *1 Nom BARRIER NPNH) ;
+MAP:r173 (@SUBJ→) TARGET (Num Nom) IF (1 TIME + Gen LINK *1 ("leat") BARRIER N LINK *1 Nom BARRIER NPNH) ;
 	## Vuosttaš logi minuvtta lei buorre áigodat Nordlysa ektui.
 
 MAP:r322 (@APP-N←) TARGET (Num Nom) IF (-2 BOS OR (PUNCT LEFT))(-1 KLASS LINK 0 Nom);
@@ -14391,7 +14391,7 @@ MAP:r198 (@→N) TARGET (Num Attr) IF (*
 
 MAP:r200 (@→N) TARGET OKTA IF (0 Acc)(*1 (N Acc) OR (A Acc) BARRIER NPNH)(NEGATE *1 Prop OR Dem BARRIER NOT-QMARK);
 
-MAP:r201 (@→N) TARGET OKTA IF (0 Gen)(*1 N OR A LINK 0 Gen OR Ill OR Loc BARRIER NPNH)(NEGATE *1 Prop OR Dem BARRIER NOT-QMARK);
+MAP:r201 (@→N) TARGET OKTA IF (0 Gen)(*1 N OR A LINK 0 Gen OR Ill OR Loc)(NEGATE *1 Prop OR Dem BARRIER NOT-QMARK);
   
 MAP:r203 (@→N) TARGET ("okta" Ess) IF (*1 (N Ess) BARRIER NPNH)(NEGATE *1 Prop OR Dem BARRIER NOT-QMARK);
   # # No example yet.
@@ -14579,7 +14579,7 @@ MAP:r253 (@Num←) TARGET Gen IF (*-1 NU
 MAP:r254 (@Num← @→N) TARGET Gen IF (*-1 NUMERALS BARRIER N OR REALCLB OR PUNCT LINK 0 Nom OR Acc LINK NOT 0 Ord OR ADVLCASE)(NEGATE 0 Num);
 	## Skuvlajagis 2001/2002 leat oktiibuot 469 oahppi suohkana vuođđoskuvllain.
 
-MAP:r255 (@Num←) TARGET (N Gen) IF (*-1 NUMERALS LINK 0 Nom LINK NOT 0 Ord BARRIER NOT-QMARK LINK *-1 go BARRIER NOT-ADV-PCLE LINK *-1 V-TRANS-ACT BARRIER NOT-QMARK) (*1 Acc BARRIER NOT-QMARK LINK *1 CC OR CLB BARRIER NOT-QMARK);
+MAP:r255 (@Num←) TARGET (N Gen) IF (*-1 (NUMERALS LINK 0 Nom LINK NOT 0 Ord) BARRIER NOT-QMARK LINK *-1 go BARRIER NOT-ADV-PCLE LINK *-1 V-TRANS-ACT BARRIER NOT-QMARK) (*1 Acc BARRIER NOT-QMARK LINK *1 CC OR CLB BARRIER NOT-QMARK);
 	## ...addet go máŋga dearvvašvuođabargiidjoavkku veahki ja fállojuvvo go...
 
 MAP:r256 (@Num←) TARGET Gen IF (*-1 CC BARRIER NOT-QMARK LINK *-1 (N Gen) BARRIER NOT-QMARK LINK *-1 NUMERALS BARRIER NP-BOUNDARY OR N OR REALCLB OR PUNCT LINK 0 Nom LINK NOT 0 Ord)(NEGATE 0 Num LINK NOT 0 Ord);
@@ -14781,7 +14781,7 @@ MAP:r321 (@APP-Pron←) TARGET (N Acc) I
   
 	# Fragment mappings
 	# -----------------
-MAP:r324 (@HNOUN) TARGET (N Prop Nom) IF (NEGATE *-1 VFIN OR ("leat" PrfPrc) OR CP)(NEGATE *1 (N Nom) BARRIER S-BOUNDARY2) (NEGATE *1 VFIN BARRIER S-BOUNDARY2)(NEGATE 1 go OR COMMA)(NEGATE *-1 COMMA BARRIER NPNH LINK *-1 ("boahtin" Nom) BARRIER VERB OR S-BOUNDARY2 LINK -1 ("bures"))(NEGATE -1 TITLE OR Mal OR Fem LINK -1 COMMA BARRIER NPNH LINK *-1 ("boahtin" Nom) BARRIER VERB OR S-BOUNDARY2 LINK -1 ("bures"))(1 EOC); 
+MAP:r324 (@HNOUN) TARGET (N Prop Nom) IF (NEGATE *-1 VFIN OR ("leat" PrfPrc) OR CP)(NEGATE *1 (N Nom) BARRIER S-BOUNDARY2) (NEGATE *1 VFIN BARRIER S-BOUNDARY2)(NEGATE 1 go OR COMMA)(NEGATE *-1 COMMA BARRIER NPNH LINK *-1 ("boahtin" Nom) BARRIER VERB OR S-BOUNDARY2 LINK -1 ("bures"))(NEGATE -1 TITLE OR Mal OR Fem LINK -1 COMMA LINK *-1 ("boahtin" Nom) BARRIER VERB OR S-BOUNDARY2 LINK -1 ("bures"))(1 EOC); 
 	## Trond Trosterud
 
 MAP:r325 (@HNOUN) TARGET N IF (-1 LEFT)(1 RIGHT);
@@ -14826,13 +14826,13 @@ MAP:r336 (@APP→Pron) TARGET (N Sg Nom)
 MAP:r337 (@VOC) TARGET Nom IF (-1 BOS OR REALCLB)(1 COMMA)(2 VFIN);
 	## Ustiban, in mun daga dutnje vearrut.
 	
-MAP:r338 (@APP-Pron←) TARGET Nom IF (*-1 Pron LINK 0 Sg2 OR Du2 Or Pl2 BARRIER NPNH LINK -1 COMMA LINK *-1 (V Sg2) OR (V Du2) OR (V Pl2) OR (Pron Sg2) OR (Pron Du2) OR (Pron Pl2) LINK NOT 0 ("leat"))(1 EOS)(0 ANIMATE)(NEGATE 0 Sg LINK *-1 V-SG3 BARRIER VERB);
+MAP:r338 (@APP-Pron←) TARGET Nom IF (*-1 (Pron LINK 0 Sg2) OR Du2 Or Pl2 BARRIER NPNH LINK -1 COMMA LINK *-1 (V Sg2) OR (V Du2) OR (V Pl2) OR (Pron Sg2) OR (Pron Du2) OR (Pron Pl2) LINK NOT 0 ("leat"))(1 EOS)(0 ANIMATE)(NEGATE 0 Sg LINK *-1 V-SG3 BARRIER VERB);
 	## Gáidet eret mu luhtte, dii verrošeaddjit! 
 	
 MAP:r339 (@VOC) TARGET Nom IF (*-1 COMMA BARRIER NOT-NP OR CC LINK *-1 (V Sg2) OR (V Du2) OR (V Pl2) OR (Pron Sg2) OR (Pron Du2) OR (Pron Pl2) LINK NOT 0 ("leat"))(1 EOS)(0 ANIMATE)(NEGATE 0 Sg LINK *-1 V-SG3 BARRIER VERB);
 	## Vuoi du, Betsaida!
 	
-MAP (@VOC) TARGET Nom IF (-1 Attr LINK -1 COMMA BARRIER NOT-NP OR CC LINK *-1 (V Sg2) OR (V Du2) OR (V Pl2) OR (Pron Sg2) OR (Pron Du2) OR (Pron Pl2) LINK NOT 0 ("leat"))(1 EOS)(0 ANIMATE)(NEGATE 0 Sg LINK *-1 V-SG3 BARRIER VERB);
+MAP (@VOC) TARGET Nom IF (-1 Attr LINK -1 COMMA OR CC LINK *-1 (V Sg2) OR (V Du2) OR (V Pl2) OR (Pron Sg2) OR (Pron Du2) OR (Pron Pl2) LINK NOT 0 ("leat"))(1 EOS)(0 ANIMATE)(NEGATE 0 Sg LINK *-1 V-SG3 BARRIER VERB);
 	
 	
 MAP:r340 (@VOC) TARGET (N Nom) IF (*-1 COMMA BARRIER NP-BOUNDARY LINK *-1 ("boahtin" Nom) BARRIER VERB OR S-BOUNDARY2 LINK -1 ("bures"))(1 EOC);
@@ -15076,7 +15076,7 @@ MAP (@←SPRED) TARGET Nom IF (*-1 COPUL
 MAP (@←SPRED) TARGET Nom IF (*-1 @ADVL BARRIER NPNHA LINK *-1 COPULAS BARRIER NPNHAI LINK 0 FMAINV LINK *-1 (@HAB) OR @ADVL OR BOS OR (@CVP) BARRIER NOT-ADV-PCLE LINK NOT 0 MO OR ADV-NOT-ARG LINK NOT -1 MO)(NEGATE *-1 ("orrut") BARRIER COPULAS LINK -1 Loc)(NEGATE 0 Attr OR Rel)(NEGATE *1 SPRED + A OR PLACE + Loc OR DOPPE BARRIER VERB OR S-BOUNDARY2 LINK *-1 COPULAS OR HAB-VERB LINK **-1 @ADVL LINK 0 TIME-MARKER OR GEASSET OR ADV-NOT-ARG OR ("mielde"))(NEGATE *1 (N Nom) OR (Pron Nom) BARRIER NPNHA OR CC OR COMMA) ; 
 	## Beassážiid leat Guovdageainnus konsearttat ja teáhterčájálmasat.
 
-MAP (@←SPRED) TARGET Nom IF (*-1 (@Pron←) BARRIER NPNHAI LINK -1 COPULAS OR HAB-VERB BARRIER NPNHAI LINK 0 FMAINV LINK **-1 (@HAB) OR @ADVL OR BOS OR (@CVP) BARRIER NOT-ADV-PCLE LINK NOT 0 ADV-NOT-ARG)(NEGATE *-1 ("orrut") BARRIER COPULAS LINK -1 Loc)(NEGATE 0 Attr)(NEGATE *1 SPRED OR PLACE + Loc OR DOPPE BARRIER VERB LINK *-1 COPULAS OR HAB-VERB LINK *-1 @ADVL LINK 0 TIME-MARKER OR GEASSET)(NEGATE *1 Nom BARRIER NPNHA) ;  
+MAP (@←SPRED) TARGET Nom IF (*-1 (@Pron←) BARRIER NPNHAI LINK -1 COPULAS OR HAB-VERB LINK 0 FMAINV LINK **-1 (@HAB) OR @ADVL OR BOS OR (@CVP) BARRIER NOT-ADV-PCLE LINK NOT 0 ADV-NOT-ARG)(NEGATE *-1 ("orrut") BARRIER COPULAS LINK -1 Loc)(NEGATE 0 Attr)(NEGATE *1 SPRED OR PLACE + Loc OR DOPPE BARRIER VERB LINK *-1 COPULAS OR HAB-VERB LINK *-1 @ADVL LINK 0 TIME-MARKER OR GEASSET)(NEGATE *1 Nom BARRIER NPNHA) ;  
 	## Mis leat alddámet ain šibihat.   
     
 MAP (@←SPRED) TARGET Nom IF (*-1 (@Pron←) BARRIER NPNHAI LINK **-1 (@HAB) OR @ADVL BARRIER NPNHAI OR (@CVP) LINK NOT 0 ADV-NOT-ARG LINK *-1 ("leat") BARRIER NOT-PCLE LINK 0 FMAINV)(NEGATE *1 Nom BARRIER NPNHA) ; 
@@ -15125,7 +15125,7 @@ MAP:r462 (@SUBJ→) TARGET ("buot") OR (
 
 MAP:r463 (@SUBJ→) TARGET ("buot") OR ("gait") IF (1 V-SG3 LINK 0 COPULAS LINK *1 PrfPrc BARRIER NOT-ADV LINK 0 IV OR Der_Pass);
 MAP:r464 (@←SUBJ) TARGET ("buot") OR ("gait") IF (*-1 V-SG3 BARRIER NOT-ADV-PCLE LINK 0 COPULAS)(*1 PrfPrc BARRIER NOT-ADV LINK 0 IV OR Der_Pass);
-MAP:r465 (@←SUBJ) TARGET ("buot") OR ("gait") IF (-1 COMMA)(1 PrfPrc BARRIER NOT-ADV LINK 0 IV OR Der_Pass);
+MAP:r465 (@←SUBJ) TARGET ("buot") OR ("gait") IF (-1 COMMA)(1 PrfPrc LINK 0 IV OR Der_Pass);
 	## Buot lea vuođđuduvvon dan áddejumis.
 	## Nu leige buot vuođđuduvvon dan áddejumis.  
 	## Sii siskkildit dán 78 riektemearrádusa, buot vuođđuduvvon dan áddejumis, almmá ovttage vuosteákka haga.
@@ -15198,7 +15198,7 @@ MAP:r3338 (@←SPRED) TARGET Nom IF (*-1
 MAP:r3339 (@←SPRED) TARGET A IF (*-1 COPULAS BARRIER SV-BOUNDARY2 OR @←SPRED LINK 0 FMAINV LINK NOT *-1 Loc BARRIER NOT-ADV-PCLE)(0 Nom OR Ess)(NEGATE 0 Attr LINK 1 N)(NEGATE *1 PrfPrc BARRIER NOT-ADV-PCLE);
 
 
-MAP:r3588 (@←SPRED) TARGET (A Nom) IF (*-1 COPULAS + Sg3 BARRIER SV-BOUNDARY2 LINK 0 FMAINV LINK -1 PLACE + Loc OR (Plc Loc) BARRIER NOT-ADV-PCLE)(NEGATE 0 Attr LINK 1 N)(NEGATE *1 PrfPrc BARRIER NOT-ADV-PCLE);
+MAP:r3588 (@←SPRED) TARGET (A Nom) IF (*-1 COPULAS + Sg3 BARRIER SV-BOUNDARY2 LINK 0 FMAINV LINK -1 PLACE + Loc OR (Plc Loc))(NEGATE 0 Attr LINK 1 N)(NEGATE *1 PrfPrc BARRIER NOT-ADV-PCLE);
 	## Dutnje lea buoret ahte don massát ovtta lahtu.
 	## Parisas lei fiinnis, muhto lei nu heajos dálki.
 
@@ -15650,10 +15650,10 @@ REMOVE:r1944 Fem IF (-2 (Prop Attr) LINK
 REMOVE:r1945 Sur IF (-1 (Prop Attr))(NEGATE -1 Mal OR Fem);
 	## Helse Nord.
 	
-SELECT:r1946 Sur IF (0 Plc LINK 0 SUBJ)(*1 VERBAL-ACTIVITY-V LINK 0 VFIN BARRIER SV-BOUNDARY2);
-SELECT:r1947 Sur IF (*1 TV LINK 0 VFIN BARRIER SV-BOUNDARY2)(0 Plc LINK 0 SUBJ);
-SELECT:r1948 Sur IF (0 Plc LINK 0 SUBJ)(*1 AUX LINK 0 VFIN BARRIER SV-BOUNDARY2 LINK *1 TV BARRIER SV-BOUNDARY2);
-SELECT:r1949 Sur IF (*-1 TV BARRIER SV-BOUNDARY2 LINK *-1 AUX LINK 0 VFIN BARRIER SV-BOUNDARY2)(0 Plc LINK 0 SUBJ);
+SELECT:r1946 Sur IF (0 Plc LINK 0 SUBJ)(*1 (VERBAL-ACTIVITY-V LINK 0 VFIN) BARRIER SV-BOUNDARY2);
+SELECT:r1947 Sur IF (*1 (TV LINK 0 VFIN) BARRIER SV-BOUNDARY2)(0 Plc LINK 0 SUBJ);
+SELECT:r1948 Sur IF (0 Plc LINK 0 SUBJ)(*1 (AUX LINK 0 VFIN) BARRIER SV-BOUNDARY2 LINK *1 TV BARRIER SV-BOUNDARY2);
+SELECT:r1949 Sur IF (*-1 TV BARRIER SV-BOUNDARY2 LINK *-1 (AUX LINK 0 VFIN) BARRIER SV-BOUNDARY2)(0 Plc LINK 0 SUBJ);
     ## Muhto lea áibbas čielggas ahte mii vuordit guorahallanproseassa luvve dan 3,1 miljovnna, dadjá Kemi.
 
 REMOVE:r1950 Plc IF (0 SUBJ LINK 0 Mal OR Fem OR Sur)(*1 DYNAMIC-V BARRIER S-BOUNDARY2 LINK 0 FMAINV OR Inf);
@@ -15832,7 +15832,7 @@ REMOVE @SUBJ→ IF (*-1 VFIN BARRIER NPN
   	# Late nominative rule
   	# --------------------
 
-SELECT:r3418 Nom IF (*-1 ("go") BARRIER NPNH LINK -1 A LINK -1 REALCOPULAS BARRIER SV-BOUNDARY2 LINK *-1C SUBJ BARRIER SV-BOUNDARY2)(0 (@COMP-CS←));
+SELECT:r3418 Nom IF (*-1 ("go") BARRIER NPNH LINK -1 A LINK -1 REALCOPULAS LINK *-1C SUBJ BARRIER SV-BOUNDARY2)(0 (@COMP-CS←));
     ## Máhttájeaddji ii leat stuorit go oahpaheaddjis.
     
 	# Fragments
@@ -16006,14 +16006,14 @@ SELECT:r3475 (@OBJ→) IF (0 TIME LINK 0
 SELECT:r3476 (@←OBJ) IF (-1 OPRED LINK -1 ("geahččat"))(0 TIME LINK 0 @ADVL);
     ##
     
-REMOVE:r3477 OBJ IF (NEGATE *-1 TIMEOBJECT-V BARRIER SV-BOUNDARY2)(*-1 TIMEADVERBIAL-V OR IV OR N-TIMEOBJECT-V LINK 0 VFIN BARRIER SV-BOUNDARY2)(0 TIME LINK 0 @ADVL);
+REMOVE:r3477 OBJ IF (NEGATE *-1 TIMEOBJECT-V BARRIER SV-BOUNDARY2)(*-1 TIMEADVERBIAL-V OR IV OR (N-TIMEOBJECT-V LINK 0 VFIN) BARRIER SV-BOUNDARY2)(0 TIME LINK 0 @ADVL);
     ## Dasto ferte nugo čuoččuhuvvon bajábealde, bidjat vuođđun ahte oamit ledje guhton badjel guokte čuohte jagi ovdal 1775 cealkámuša, ja ahte ...
     ## Nissonolbmuid vástevaš váldoprošeakta bisttii jagi ja loahpahuvvui čakčat 1997.
     
-REMOVE:r3478 OBJ IF (NEGATE *-1 TIMEOBJECT-V BARRIER SV-BOUNDARY2)(*1 TIMEADVERBIAL-V OR IV OR N-TIMEOBJECT-V LINK 0 VFIN BARRIER SV-BOUNDARY2)(0 TIME LINK 0 @ADVL);
+REMOVE:r3478 OBJ IF (NEGATE *-1 TIMEOBJECT-V BARRIER SV-BOUNDARY2)(*1 TIMEADVERBIAL-V OR IV OR (N-TIMEOBJECT-V LINK 0 VFIN) BARRIER SV-BOUNDARY2)(0 TIME LINK 0 @ADVL);
     ## Daid vahkuid bivde luosa Geavgŋá vuolde.
     
-REMOVE:r3479 OBJ IF (*-1 TIMEADVERBIAL-V OR IV OR N-TIMEOBJECT-V LINK 0 PrfPrc OR Inf BARRIER SV-BOUNDARY2 LINK *-1 AUX BARRIER SV-BOUNDARY2)(0 TIME LINK 0 @ADVL); 
+REMOVE:r3479 OBJ IF (*-1 TIMEADVERBIAL-V OR IV OR (N-TIMEOBJECT-V LINK 0 PrfPrc) OR Inf BARRIER SV-BOUNDARY2 LINK *-1 AUX BARRIER SV-BOUNDARY2)(0 TIME LINK 0 @ADVL); 
     ## Visot dát joavkkut lea eallán Ruoŧas guhkes áiggi ja leat danin oassi ruoŧa kulturárbbis. 
     ## Joavku láve maid rahčat bures, ja doaivvun maid dainna nagodit joatkit nuppe jagi.
     
@@ -16021,7 +16021,7 @@ REMOVE:r3480 OBJ IF (*-1C OBJ OR OPRED B
 REMOVE:r3481 OBJ IF (-1 (@CNP) LINK -1 RCmpnd OR (Acc @ADVL) LINK *-1 OBJ OR OPRED BARRIER SV-BOUNDARY2)(0 TIME LINK 0 @ADVL);
     ## Bivdu, eandálii beatnagiin bivdin, sáhttá hehttet boazobarggu čohkken- ja johtináiggi.
     
-SELECT:r3482 (@←OBJ) IF (*-1 TIMEOBJECT-V BARRIER SV-BOUNDARY2 LINK NOT 0 Actio)(0 TIME LINK 0 @ADVL)(NEGATE *1 OBJ BARRIER SV-BOUNDARY2 LINK *1 TIMEOBJECT-V LINK 0 VFIN BARRIER SV-BOUNDARY2 LINK NOT *-1 OBJ BARRIER SV-BOUNDARY2);
+SELECT:r3482 (@←OBJ) IF (*-1 TIMEOBJECT-V BARRIER SV-BOUNDARY2 LINK NOT 0 Actio)(0 TIME LINK 0 @ADVL)(NEGATE *1 OBJ BARRIER SV-BOUNDARY2 LINK *1 (TIMEOBJECT-V LINK 0 VFIN) BARRIER SV-BOUNDARY2 LINK NOT *-1 OBJ BARRIER SV-BOUNDARY2);
     ## Guotte min morrašis, nu ahte mii nagodivččiimet gierdat morraša beivviid.
     ## Dás eai leat váikkuhusat iešguđet ekonomalaš váikkuhangaskaomiide ja dat lea dušše heiveheapmi stáhtabušehttii mii čuovvu kaleanddarjagi.
     
@@ -16038,19 +16038,19 @@ SELECT:r3487 Acc IF (-1 (@CNP) LINK -1C
     
     # ROUTE 
     
-SELECT:r3489 (@←OBJ) IF (*-1 ROUTEOBJECT-V BARRIER SV-BOUNDARY2)(0 ROUTE LINK 0 @ADVL)(NEGATE *1 OBJ BARRIER SV-BOUNDARY2 LINK *1 ROUTEOBJECT-V LINK 0 VFIN BARRIER SV-BOUNDARY2 LINK NOT *-1 OBJ BARRIER SV-BOUNDARY2);
+SELECT:r3489 (@←OBJ) IF (*-1 ROUTEOBJECT-V BARRIER SV-BOUNDARY2)(0 ROUTE LINK 0 @ADVL)(NEGATE *1 OBJ BARRIER SV-BOUNDARY2 LINK *1 (ROUTEOBJECT-V LINK 0 VFIN) BARRIER SV-BOUNDARY2 LINK NOT *-1 OBJ BARRIER SV-BOUNDARY2);
     ## Olusat lebbo biktasiidiset geainnu ala, earát čuhppe lastaovssiid gittiin ja duorgastedje geainnu.
     ## Geahča, mun vuolggahan áirasan du ovddabeale, son ráhkada dutnje geainnu.
     ## Don leat buot vanhurskkisvuođa vašálaš, dievva juohkelágán gávvilvuođas ja bahávuođas, it go goassege áiggo heaitit botnjamis Hearrá njulges geainnuid?
     
-SELECT:r3490 (@OBJ→) IF (-1 ("vejolašvuohta") LINK -1 ("leat"))(0 ROUTE LINK 0 @ADVL)(*1 ROUTEOBJECT-V LINK 0 VFIN BARRIER SV-BOUNDARY2 LINK NOT *-1 OBJ BARRIER SV-BOUNDARY2);
+SELECT:r3490 (@OBJ→) IF (-1 ("vejolašvuohta") LINK -1 ("leat"))(0 ROUTE LINK 0 @ADVL)(*1 (ROUTEOBJECT-V LINK 0 VFIN) BARRIER SV-BOUNDARY2 LINK NOT *-1 OBJ BARRIER SV-BOUNDARY2);
     ## 
     # No hits 800409
 
-REMOVE:r3491 OBJ IF (*-1 ROUTEADVERBIAL-V OR IV OR MOVEMENT-V LINK 0 VFIN BARRIER SV-BOUNDARY2)(0 ROUTE LINK 0 @ADVL); 
+REMOVE:r3491 OBJ IF (*-1 ROUTEADVERBIAL-V OR IV OR (MOVEMENT-V LINK 0 VFIN) BARRIER SV-BOUNDARY2)(0 ROUTE LINK 0 @ADVL); 
     ## 
     
-REMOVE:r3492 OBJ IF (*-1 ROUTEADVERBIAL-V OR IV OR MOVEMENT-V LINK 0 PrfPrc BARRIER SV-BOUNDARY2 LINK *-1 ("leat") LINK 0 VFIN)(0 ROUTE LINK 0 @ADVL); 
+REMOVE:r3492 OBJ IF (*-1 ROUTEADVERBIAL-V OR IV OR (MOVEMENT-V LINK 0 PrfPrc) BARRIER SV-BOUNDARY2 LINK *-1 ("leat") LINK 0 VFIN)(0 ROUTE LINK 0 @ADVL); 
     ##     
 
 	# Genitive
@@ -16228,7 +16228,7 @@ SELECT:r3558 (Indef Gen @→N) IF (*1C (
 
 	# Case
 
-REMOVE:r3559 Nom IF (*1C V-NOT-AUX LINK NOT 0C (PrfPrc @→N) BARRIER S-BOUNDARY2 OR PUNCT LINK *1 FMAINV BARRIER NOT-ADV-PCLE LINK *1C NP-HEAD-NOM BARRIER S-BOUNDARY2 OR ("dego") OR ("dugo") LINK NOT 0 Interr)(NEGATE -1 COMMA LINK -1 (@SPRED));
+REMOVE:r3559 Nom IF (*1C V-NOT-AUX LINK NOT 0C (PrfPrc @→N) OR PUNCT LINK *1 FMAINV BARRIER NOT-ADV-PCLE LINK *1C NP-HEAD-NOM BARRIER S-BOUNDARY2 OR ("dego") OR ("dugo") LINK NOT 0 Interr)(NEGATE -1 COMMA LINK -1 (@SPRED));
 	
 	## Dakkár buođus gáržo guolli juo nu ahte dat deaddil joddu sisa.
 
--- apertium-sme-nob-0.5.0/apertium-sme-nob.sme-nob.lex.orig	2012-09-18 09:33:22.000000000 +0200
+++ apertium-sme-nob-0.5.0/apertium-sme-nob.sme-nob.lex	2013-06-29 23:05:39.053141340 +0200
@@ -1954,7 +1954,7 @@ SUBSTITUTE ("lohkat") ("lohkat:1")
 		(1 ("ahte") OR (Refl Acc) OR (Refl Loc) OR PrfPrc)
 		OR
 		(*1 FMAINV OR Actio OR PrfPrc OR Inf BARRIER S-BOUNDARY OR ("galle") OR ("man"))
-	   ) (NEGATE *0 OBJ LINK 0 READABLE BARRIER S-BOUNDARY)
+	   ) (NEGATE *0 (OBJ LINK 0 READABLE) BARRIER S-BOUNDARY)
 	   (NEGATE *0 ("jitnosit") OR ("hiđis" Adv) OR ("jaskat" Adv) BARRIER S-BOUNDARY OR V);
 	## Ovddeš bargi Yle Sámi Radios, Ánne Risten Juuso, lohká ahte Gárasavvonis livčče eará latnja leamaš Yle Sámi radio doaimmahussii. -  Den tidligere arbeideren Yle på Samelands Radio, Ánne Risten Juuso, hun sier at ...
 	## Son lohká máddin Sámis lea sámit garrasabbot deddon dahje vealahuvvon go davvin. - Han sier sørfra har Sameland samer hardere trykt eller berøvd nordpå.
@@ -1964,7 +1964,7 @@ SUBSTITUTE ("lohkat") ("lohkat:1")
 	
 SUBSTITUTE ("lohkat") ("lohkat:2")
 	TARGET ("lohkat" V)
-	IF (*1 (@←OBJ) LINK NOT 0 TEXT BARRIER S-BOUNDARY);
+	IF (*1 (@←OBJ) LINK NOT 0 TEXT);
 SUBSTITUTE ("lohkat") ("lohkat:2")
 	TARGET ("lohkat" V)
 	IF (1 ("galle") OR ("man") LINK 1 QUANT-PRON LINK *1 (@←OBJ) BARRIER NPNH);
